### PR TITLE
Task/store missing put 3 mine resource 

### PR DESCRIPTION
**Pull request by Jsyro**
Tackling the mine put and post endpoints... they have needed some TLC for a while, so it took some changes, but after updating some of the tests to match some error formats. It looks good, Please try and break this.

### CODE REVIEW

File: frontend/src/actionCreators/mineActionCreator.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ export const createMineRecord = (payload) => (dispatch) => {
     .then(createMineTypeRequests(payload, dispatch, reducerTypes.CREATE_MINE_RECORD))
     .then((response) => {
       notification.success({
-        message: `Successfully created: ${payload.name}`,
+        message: `Successfully created: ${payload.mine_name}`,
```
Comment (nayr974) [inline frontend/src/actionCreators/mineActionCreator.js]: yes

File: frontend/src/components/mine/MineHeader.js [hunk 2]
```diff
@@ -113,13 +113,13 @@ export class MineHeader extends Component {
...
-      name: mine.mine_name ? mine.mine_name : null,
+      mine_name: mine.mine_name,
       latitude: mine.mine_location ? mine.mine_location.latitude : null,
       longitude: mine.mine_location ? mine.mine_location.longitude : null,
       mine_status: mine.mine_status[0] ? mine.mine_status[0].status_values : null,
       major_mine_ind: mine.major_mine_ind ? mine.major_mine_ind : false,
-      mine_region: mine.region_code ? mine.region_code : null,
-      note: mine.mine_note ? mine.mine_note : null,
+      mine_region: mine.region_code,
+      mine_note: mine.mine_note,
```
Comment (nayr974) [inline frontend/src/components/mine/MineHeader.js]: yes

File: python-backend/app/api/mines/location/models/mine_location.py [hunk 3]
```diff
@@ -2,14 +2,17 @@
...
 from geoalchemy2 import Geometry
 from ....utils.models_mixins import AuditMixin, Base
 from app.extensions import db, cache
 
 
 class MineLocation(AuditMixin, Base):
     __tablename__ = "mine_location"
-    mine_location_guid = db.Column(UUID(as_uuid=True), primary_key=True)
+    mine_location_guid = db.Column(
+        UUID(as_uuid=True), primary_key=True, server_default=FetchedValue())
```
Comment (nayr974) [inline python-backend/app/api/mines/location/models/mine_location.py]: yes

File: python-backend/app/api/mines/mine/models/mine.py [hunk 4]
```diff
@@ -18,7 +18,7 @@
 
 class Mine(AuditMixin, Base):
     __tablename__ = 'mine'
-    mine_guid = db.Column(UUID(as_uuid=True), primary_key=True)
+    mine_guid = db.Column(UUID(as_uuid=True), primary_key=True, server_default=FetchedValue())
```
Comment (nayr974) [inline python-backend/app/api/mines/mine/models/mine.py]: yes

File: python-backend/app/api/mines/mine/models/mine.py [hunk 5]
```diff
@@ -226,3 +226,11 @@ def validate_mine_no(self, key, mine_no):
         if mine_no and len(mine_no) > 10:
             raise AssertionError('Mine number must not exceed 10 characters.')
         return mine_no
+
+    @validates('mine_region')
+    def validate_mine_region(self, key, mine_region):
```
Comment (nayr974) [inline python-backend/app/api/mines/mine/models/mine.py]: yes

File: python-backend/app/api/mines/mine/resources/mine.py [hunk 6]
```diff
@@ -197,46 +217,42 @@ def mine_operation_code_processor(self, mine_status, index):
...
             if not mine_status_xref:
-                self.raise_error(
-                    400,
-                    'Error: Invalid status_code, reason_code, and sub_reason_code combination., '
-                ), 400
+                raise BadRequest(
+                    'Invalid status_code, reason_code, and sub_reason_code combination.')
 
-            existing_status = MineStatus.find_by_mine_guid(mine_guid)
+            existing_status = mine.mine_status
```
Comment (nayr974) [inline python-backend/app/api/mines/mine/resources/mine.py]: Are you sure this works? Curious why previously we had to find it.

Comment (Jsyro) [reply python-backend/app/api/mines/mine/resources/mine.py]: This function was accepting the mine_guid. I changed it to accept the mine object so i could assign the relationship so don't need to save the mine before saving the status.

It works.

**Pull request merged**